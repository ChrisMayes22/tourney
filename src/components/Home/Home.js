import React from 'react';
import classes from './Home.css';
import SubmitButton from '../SubmitButton/SubmitButton';

const home = (props) => {
    return(
        <div className={classes.background}>
            <nav className={classes.ui}>
                <h3>How many judges are joining us...?</h3>
                <input type="number"></input>
                <SubmitButton/>
            </nav>
        </div>
    )
}

export default home;