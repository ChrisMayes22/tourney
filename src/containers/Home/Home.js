import React, { Component } from 'react';
import classes from './Home.css';
import SubmitButton from '../../components/SubmitButton/SubmitButton';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux'; 
import { setPlayers } from '../../constants/actions';
import * as urls from '../../constants/urls';

export class Home extends Component {

    render(){
        return(
            <div className={classes.background}>
                <nav className={classes.ui}>
                    <h3>How many judges are joining us...?</h3>
                    <input 
                        type="number"
                        onChange={(event) => this.props.onSetPlayers(Number(event.target.value))}></input>
                        {(this.props.players === 3 || this.props.players === 4)  ? 
                            <Link to={urls.UPLOAD_PAGE} id='uploadLink'>
                                <SubmitButton children={"Let's begin!"} id="beginButton"/>
                            </Link> : 
                            <h3>This game is designed for THREE or FOUR players</h3>}
                </nav>
            </div>
        )
    }
}

const mapStateToProps = state => {
    return {
        players: state.players
    }
};

const mapDispatchToProps = dispatch => {
    return {
        onSetPlayers: (num) => dispatch(setPlayers(num))
    }
 }

export default connect(mapStateToProps, mapDispatchToProps)(Home);