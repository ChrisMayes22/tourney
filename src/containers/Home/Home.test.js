import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter(), disableLifecycleMethods: true });

import React from 'react';
import { shallow } from 'enzyme';

import { Home } from './Home';

var wrapper = null;

describe('Basic render tests for component', function(){
    test('renders as expected', () => {
      wrapper = shallow(<Home/>)
    
      expect(wrapper).toMatchSnapshot();
    })
})

describe('Conditional Rendering Tests for unconnected component', function(){
  beforeEach(function(){
    const props = {players: 3}
    wrapper = shallow(<Home {...props} />)
  })
  describe('when players === 3 or players === 4', function(){
    test('Link and SubmitButton render once each', function(){
      expect(wrapper.exists('Link')).toBe(true);
      expect(wrapper.exists('#beginButton')).toBe(true);

      wrapper.setProps({players: 4});

      expect(wrapper.exists('Link')).toBe(true);
      expect(wrapper.exists('#beginButton')).toBe(true);
    })
  })
  describe('when players !== 3 and players !== 4', function(){
    test('Link and SubmitButton do not render', function(){
      wrapper.setProps({players: 0});
      
      expect(wrapper.exists('Link')).toBe(false);
      expect(wrapper.exists('#beginButton')).toBe(false);

      wrapper.setProps({players: 5})

      expect(wrapper.exists('Link')).toBe(false);
      expect(wrapper.exists('#beginButton')).toBe(false);
    })
  })
})
