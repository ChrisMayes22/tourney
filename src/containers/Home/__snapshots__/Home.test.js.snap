// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic render tests for component renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <nav>
        <h3>
          How many judges are joining us...?
        </h3>
        <input
          onChange={[Function]}
          type="number"
        />
        <h3>
          This game is designed for TWO to FOUR players
        </h3>
      </nav>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            How many judges are joining us...?
          </h3>,
          <input
            onChange={[Function]}
            type="number"
          />,
          <h3>
            This game is designed for TWO to FOUR players
          </h3>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "How many judges are joining us...?",
          },
          "ref": null,
          "rendered": "How many judges are joining us...?",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "type": "number",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This game is designed for TWO to FOUR players",
          },
          "ref": null,
          "rendered": "This game is designed for TWO to FOUR players",
          "type": "h3",
        },
      ],
      "type": "nav",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav>
          <h3>
            How many judges are joining us...?
          </h3>
          <input
            onChange={[Function]}
            type="number"
          />
          <h3>
            This game is designed for TWO to FOUR players
          </h3>
        </nav>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              How many judges are joining us...?
            </h3>,
            <input
              onChange={[Function]}
              type="number"
            />,
            <h3>
              This game is designed for TWO to FOUR players
            </h3>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "How many judges are joining us...?",
            },
            "ref": null,
            "rendered": "How many judges are joining us...?",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "type": "number",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "This game is designed for TWO to FOUR players",
            },
            "ref": null,
            "rendered": "This game is designed for TWO to FOUR players",
            "type": "h3",
          },
        ],
        "type": "nav",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
